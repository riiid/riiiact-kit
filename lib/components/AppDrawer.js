import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Drawer from 'material-ui/Drawer';
import MenuItem from 'material-ui/MenuItem';
import {spacing, typography} from 'material-ui/styles';
import {cyan500} from 'material-ui/styles/colors';
import {VERSION, TARGET} from 'config';

const styles = {
  logo: {
    cursor: 'pointer',
    fontSize: 16,
    color: typography.textFullWhite,
    lineHeight: `${spacing.desktopKeylineIncrement}px`,
    fontWeight: typography.fontWeightLight,
    backgroundColor: cyan500,
    paddingLeft: spacing.desktopGutter,
    marginBottom: 8
  }
};

class AppDrawer extends Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  render() {
    const {open, onToggle} = this.props;
    return (
      <Drawer docked={false} open={open}
        onRequestChange={onToggle}>
        <div style={styles.logo}>
          v{VERSION}{TARGET === 'development' ? '-dev' : ''}
        </div>
        <MenuItem disabled={true}>user@email.com</MenuItem>
      </Drawer>
    );
  }
}

AppDrawer.displayName = 'AppDrawer';
AppDrawer.propTypes = {
  open: PropTypes.bool,
  onToggle: PropTypes.func
};

export default AppDrawer;
