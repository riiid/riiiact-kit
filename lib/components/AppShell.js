import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {firebase} from 'api';
import Row from 'react-bootstrap/lib/Row';
import Col from 'react-bootstrap/lib/Col';
import Alert from 'react-bootstrap/lib/Alert';

class AppShell extends Component {
  getChildContext() {
    return {db: firebase.db, auth: firebase.auth};
  }

  _children() {
    const {children} = this.props;

    if (firebase.db === null || firebase.auth === null) {
      return (
        <div className="container">
          <Row>
            <Col xs={10} sm={8} md={8} xsOffset={1} smOffset={2} mdOffset={2}>
              <Alert bsStyle="danger">
                Cannot initialize <code>App</code>.
                Did you update <code>build/config.json</code> properly?
              </Alert>
            </Col>
          </Row>
        </div>
      );
    }

    return children;
  }

  render() {
    return this._children();
  }
}

AppShell.displayName = 'AppShell';
AppShell.propTypes = {
  children: PropTypes.node
};
AppShell.childContextTypes = {
  db: PropTypes.object,
  auth: PropTypes.object
};

export default AppShell;
