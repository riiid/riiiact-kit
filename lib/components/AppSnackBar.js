import React from 'react';
import PropTypes from 'prop-types';
import Snackbar from 'material-ui/Snackbar';

const AppSnackbar = props => {
  return <Snackbar {...props}/>;
};

AppSnackbar.displayName = 'AppSnackBar';
AppSnackbar.propTypes = {
  open: PropTypes.bool,
  message: PropTypes.string
};
AppSnackbar.defaultProps = {
  open: false,
  message: ''
};

export default AppSnackbar;
