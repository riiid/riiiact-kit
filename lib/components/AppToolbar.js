import React, {Component, PropTypes} from 'react';
import AppBar from 'material-ui/AppBar';
import IconButton from 'material-ui/IconButton';
import IconMenu from 'material-ui/IconMenu';
import MenuItem from 'material-ui/MenuItem';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';
import DescIcon from 'material-ui/svg-icons/action/description';
import ChartIcon from 'material-ui/svg-icons/editor/show-chart';
import ClassIcon from 'material-ui/svg-icons/action/class';
import FeedbackIcon from 'material-ui/svg-icons/action/feedback';

class AppToolbar extends Component {

  constructor(props) {
    super(props);
    this.state = {
    };
  }

  _route(index) {
    const {onOpenSnackbar} = this.props;
    const {router} = this.context;
    switch (index) {
      case 0:
      case 1:
      case 2:
        return router.push('menu');
      case 3:
        return onOpenSnackbar();
      case 100:
        return router.push('/');
      default:
        break;
    }
  }

  _menu() {
    return (
      <IconMenu iconButtonElement={<IconButton><MoreVertIcon /></IconButton>}
        targetOrigin={{horizontal: 'right', vertical: 'top'}}
        anchorOrigin={{horizontal: 'right', vertical: 'top'}}>
      <MenuItem primaryText="Menus" disabled={true} />
      <MenuItem primaryText="menu 1"
        leftIcon={<DescIcon/>}
        onTouchTap={this._route.bind(this, 0)}/>
      <MenuItem primaryText="menu 2"
        leftIcon={<ChartIcon/>}
        onTouchTap={this._route.bind(this, 1)}/>
      <MenuItem primaryText="menu 3"
        leftIcon={<ClassIcon/>}
        onTouchTap={this._route.bind(this, 2)}/>
      <MenuItem primaryText="show snackbar"
        leftIcon={<FeedbackIcon/>}
        onTouchTap={this._route.bind(this, 3)}/>
      </IconMenu>
    );
  }

  render() {
    const {className, onToggleDrawer} = this.props;
    return (
      <AppBar style={{position: 'fixed', top: 0, zIndex: 1101}}
        className={className}
        onLeftIconButtonTouchTap={onToggleDrawer}
        onTitleTouchTap={this._route.bind(this, 100)}
        iconElementRight={this._menu()}
        title="Riiiact-Kit"
        titleStyle={{cursor: 'pointer'}}/>
    );
  }
}

AppToolbar.displayName = 'AppToolbar';
AppToolbar.contextTypes = {
  router: PropTypes.object.isRequired
};
AppToolbar.propTypes = {
  className: PropTypes.string,
  onToggleDrawer: PropTypes.func,
  onOpenSnackbar: PropTypes.func
};

export default AppToolbar;
