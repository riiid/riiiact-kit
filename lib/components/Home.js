import React, {Component} from 'react';
import Row from 'react-bootstrap/lib/Row';
import Col from 'react-bootstrap/lib/Col';
import {TITLE, DEPENDENCIES} from 'config';
import map from 'lodash/map';
import {Card, CardTitle, CardText} from 'material-ui/Card';

class Home extends Component {
  render() {
    return (
      <Row>
        <Col xs={10} sm={8} md={8} xsOffset={1} smOffset={2} mdOffset={2}>
          <Card>
            <CardTitle title={TITLE} subtitle="powered by"/>
            <CardText>
              <ul style={{marginTop: 0}}>
              {
                map(DEPENDENCIES, (v, k) => {
                  return (
                    <li key={k}><code>{k}</code> : <code>{v}</code></li>
                  );
                })
              }
              </ul>
            </CardText>
          </Card>
        </Col>
      </Row>
    );
  }
}

Home.displayName = 'Home';

export default Home;
