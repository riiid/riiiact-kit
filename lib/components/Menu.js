import React, {Component} from 'react';
import {Row, Col} from 'react-bootstrap';
import {Observable} from 'rx';
import {keys} from 'lodash';

class Menu extends Component {

  constructor(props) {
    super(props);
    this.state = {
      totalViews: 0,
      currentViews: 0
    };
  }

  componentDidMount() {
    console.log('mount?');
    this._count();
    this._views();
  }

  _count() {
    const {db} = this.context;
    db.ref('stats').child('totalViews').transaction$(current => {
      if (current === null) {
        return 1;
      }
      return current + 1;
    })
    .subscribe(() => {
    }, err => {
      console.error(err);
    });
  }

  _views() {
    const {db} = this.context;
    const tv$ = db.ref('stats').child('totalViews').observe$('value')
      .map(snapshot => snapshot.val());
    const cv$ = db.ref('online-users').observe$('value')
      .map(snapshot => snapshot.val());

    this.viewsSubs = Observable.combineLatest(tv$, cv$, (tv, cv) => {
      return {
        totalViews: tv,
        currentViews: keys(cv).length
      };
    })
    .subscribe(result => this.setState(result));
  }

  componentWillUnmount() {
    this.viewsSubs.dispose();
  }

  render() {
    const {totalViews, currentViews} = this.state;
    return (
      <Row>
        <Col className="card"
          xs={10} sm={8} md={8} xsOffset={1} smOffset={2} mdOffset={2}>
          <div className="clearfix">
            <h4 className="pull-left">Hello, Folks</h4>
            <div className="pull-right">
              <span>{totalViews} Views</span>,
              &nbsp;&nbsp;
              <span>{currentViews} Current</span>
            </div>
          </div>
        </Col>
      </Row>
    );
  }
}

Menu.displayName = 'Menu';
Menu.contextTypes = {
  router: React.PropTypes.object,
  db: React.PropTypes.object
};

export default Menu;
