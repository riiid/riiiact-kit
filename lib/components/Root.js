import React from 'react';
import router from 'components/Router';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import {Provider} from 'react-redux';
import configureStore from 'store';

const store = configureStore();
const Root = () => {
  return (
    <Provider store={store}>
      <MuiThemeProvider muiTheme={getMuiTheme()}>
        {router(store)}
      </MuiThemeProvider>
    </Provider>
  );
};

Root.displayName = 'Root';

export default Root;
