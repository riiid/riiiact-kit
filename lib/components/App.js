import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import cn from 'classnames';
import {toggleDrawer, showSnackbar, hideSnackbar} from 'actions/app';
import AppToolbar from 'components/AppToolbar';
import AppDrawer from 'components/AppDrawer';
import AppSnackBar from 'components/AppSnackBar';
import LinearProgress from 'material-ui/LinearProgress';

export class App extends Component {
  _children() {
    const {children} = this.props;
    const {user} = this.props;

    if (user === null) {
      return;
    }

    return children;
  }

  _toggleDrawer() {
    const {toggleDrawer} = this.props;
    toggleDrawer();
  }

  _openSnackbar() {
    const {showSnackbar} = this.props;
    showSnackbar('Hello there?');
  }

  _closeSnackbar() {
    const {hideSnackbar} = this.props;
    hideSnackbar();
  }

  _loading() {
    const {muiTheme: {appBar: {height}}} = this.context;
    const {toolbar, loading} = this.props;
    if (loading) {
      const top = toolbar ? height : 0;
      return (
        <div style={{top, position: 'fixed', width: '100%'}}
          className="app-progress">
          <LinearProgress mode="indeterminate"/>
        </div>
      );
    }
  }

  render() {
    const {toolbar, drawerOpen, snackbar} = this.props;
    return (
      <div>
        <AppToolbar className={cn({hide: !toolbar})}
          onToggleDrawer={this._toggleDrawer.bind(this)}
          onOpenSnackbar={this._openSnackbar.bind(this)}/>
        {this._loading()}
        <AppDrawer open={drawerOpen} onToggle={this._toggleDrawer.bind(this)}/>
        <div className={cn({container: toolbar})}>
          {this._children()}
        </div>
        <AppSnackBar {...snackbar}
          onRequestClose={this._closeSnackbar.bind(this)}/>
        <br/>
      </div>
    );
  }
}

App.displayName = 'App';
App.contextTypes = {
  muiTheme: PropTypes.object
};
App.propTypes = {
  drawerOpen: PropTypes.bool,
  snackbar: PropTypes.object,
  toggleDrawer: PropTypes.func,
  showSnackbar: PropTypes.func,
  hideSnackbar: PropTypes.func,
  toolbar: PropTypes.bool,
  loading: PropTypes.bool,
  user: PropTypes.object,
  children: PropTypes.node
};
App.defaultProps = {
  drawerOpen: false
};

const mapStateToProps = state => {
  const {app, auth} = state;
  return {
    drawerOpen: Boolean(app.drawerOpen),
    loading: Boolean(app.progress),
    snackbar: app.snackbar,
    user: auth.user,
    toolbar: app.toolbar
  };
};

const mapDispatchToProps = {
  toggleDrawer,
  showSnackbar,
  hideSnackbar
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
