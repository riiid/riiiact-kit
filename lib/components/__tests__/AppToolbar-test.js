import React from 'react';
import {shallow, mount} from 'enzyme';
import TestUtils from 'react-addons-test-utils';
import {AppToolbar} from 'components/AppToolbar';

describe('components/AppToolbar', () => {

  it('should exists', () => {
    const wrapper = shallow(<AppToolbar/>, __OPTIONS);

    expect(wrapper).not.toBeNull();
    wrapper.unmount();
  });

  it('should exists', () => {
    __OPTIONS.context.router.push.mockClear();

    const openSnackbar = jest.fn();
    const wrapper = shallow(
      <AppToolbar onOpenSnackbar={openSnackbar}/>,
        __OPTIONS
    );
    const targetFn = wrapper.instance()._route.bind(wrapper.instance());
    targetFn(0);
    targetFn(2);
    targetFn(3);
    targetFn(100);
    targetFn(-1);

    expect(openSnackbar).toHaveBeenCalled();
    expect(__OPTIONS.context.router.push.mock.calls)
    .toEqual([['menu'], ['menu'], ['/']]);
    wrapper.unmount();
  });

  it('should have 2 button', () => {
    const wrapper = mount(<AppToolbar/>, __OPTIONS);

    expect(wrapper.find('button').length).toBe(2);
    wrapper.unmount();
  });

  it('should have IconMenu', () => {
    const wrapper = mount(<AppToolbar/>, __OPTIONS);

    expect(wrapper.find('IconMenu').length).toBe(1);
    wrapper.unmount();
  });

  it('should show menu items when IconButton touched', () => {
    const wrapper = mount(<AppToolbar/>, __OPTIONS);

    expect(wrapper.find('RenderToLayer').prop('open')).not.toBeTruthy();
    expect(document.body.querySelector('div:nth-child(2)')).toBeNull();

    wrapper.find('IconMenu').find('IconButton').simulate('touchTap');

    expect(wrapper.find('RenderToLayer').prop('open')).toBeTruthy();
    expect(document.body.querySelector('div:nth-child(2)').textContent)
    .toBe('Menusmenu 1menu 2menu 3show snackbar');

    wrapper.unmount();
  });
});

