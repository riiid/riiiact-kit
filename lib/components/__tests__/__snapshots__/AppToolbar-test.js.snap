// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AppToolbar render 1`] = `
<div
  className={undefined}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#00bcd4",
      "borderRadius": "0px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "flex",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "paddingLeft": 24,
      "paddingRight": 24,
      "position": "fixed",
      "top": 0,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "100%",
      "zIndex": 1101,
    }
  }
>
  <button
    disabled={false}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onTouchStart={[Function]}
    onTouchTap={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "border": 10,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 0,
        "fontWeight": "inherit",
        "height": 48,
        "margin": 0,
        "marginLeft": -16,
        "marginRight": 8,
        "marginTop": 8,
        "muiPrepared": true,
        "outline": "none",
        "overflow": "visible",
        "padding": 12,
        "position": "relative",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
        "width": 48,
        "zIndex": 1,
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "#ffffff",
            "display": "inline-block",
            "fill": "#ffffff",
            "height": 24,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </div>
  </button>
  <h1
    onTouchTap={[Function]}
    style={
      Object {
        "boxFlex": 1,
        "color": "#ffffff",
        "cursor": "pointer",
        "flex": "1",
        "fontSize": 24,
        "fontWeight": 400,
        "height": 64,
        "letterSpacing": 0,
        "lineHeight": "64px",
        "margin": 0,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingTop": 0,
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }
  >
    
  </h1>
  <div
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": -16,
        "marginTop": 8,
        "muiPrepared": true,
      }
    }
  >
    <div
      className={undefined}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "display": "inline-block",
          "muiPrepared": true,
          "position": "relative",
        }
      }
    >
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "border": 10,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 0,
            "fontWeight": "inherit",
            "height": 48,
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "visible",
            "padding": 12,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": 48,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "#ffffff",
                "display": "inline-block",
                "fill": "#ffffff",
                "height": 24,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "width": 24,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </div>
      </button>
      <div
        style={
          Object {
            "display": "none",
          }
        }
      />
    </div>
  </div>
</div>
`;
